src/GameFace/Struct/UserProfile.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace GameFace.Struct
+{
+    public static class Helper
+    {
+        public static List<UserRecords> AddElement(List<UserRecords> list, UserRecords element)
+        {
+            var found = false;
+            if (list.Count == 0)
+            {
+                list.Add(element);
+            }
+            else
+            {
+                for (var i= 0; i < list.Count; i++)
+                {
+                    if (!list[i].Task.Equals(element.Task)) continue;
+                    var newrecord = new UserRecords(list[i].Task, list[i].NoPerforms + 1, list[i].Value + element.Value);
+                    list.Remove(list[i]);
+                    list.Add(newrecord);
+                    found = true;
+                }
+                if (!found)
+                {
+                    list.Add(element);
+                }
+            }
+            
+            return list;
+        }
+
+
+        public static List<SprintRecords> DevideBySprints(List<UserRecordsDate> list)
+        {
+            var date = list[0].Date;
+            int quarterNumber = (date.Month - 1) / 3 + 1;
+            var firstDayOfQuarter = new DateTime(date.Year, (quarterNumber - 1) * 3 + 1, 1);
+            var lastDayOfQuarter = firstDayOfQuarter.AddMonths(3).AddDays(-1);
+
+            var listtemp = new List<UserRecords>();
+            var sprint = new List<SprintRecords>();
+            foreach (var record in list)
+            {
+                if (firstDayOfQuarter < record.Date && record.Date < lastDayOfQuarter)
+                {
+                    listtemp = AddElement(listtemp, new UserRecords(record.Task,1,record.Value));
+                }
+                else
+                {
+                    sprint.Add(new SprintRecords(firstDayOfQuarter, lastDayOfQuarter, listtemp));
+                    listtemp = new List<UserRecords> { new UserRecords(record.Task, 1, record.Value) };
+                    date = record.Date;
+                    quarterNumber = (date.Month - 1) / 3 + 1;
+                    firstDayOfQuarter = new DateTime(date.Year, (quarterNumber - 1) * 3 + 1, 1);
+                    lastDayOfQuarter = firstDayOfQuarter.AddMonths(3).AddDays(-1);
+                }
+            }
+            if (listtemp.Count > 0)
+            {
+                sprint.Add(new SprintRecords(firstDayOfQuarter, lastDayOfQuarter, listtemp));
+            }
+            return sprint;
+        }
```

Comment:
GroupBy groups objects together based on a key. In our case, the key is made up of the first day and last day of the quarter.
Select transforms an object into something else. In this case, we transform the groups of UserRecordDates to SprintRecords.